src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -308,24 +314,42 @@ impl FlowServiceImpl {
 
                 // When dependent flow batching rule is enabled, schedule dependent update
                 if let Some(batching_rule) = maybe_batching_rule {
-                    let dependent_flow_key = FlowKeyDataset::new(
-                        dependent_dataset_id.clone(),
-                        DatasetFlowType::ExecuteTransform,
-                    )
-                    .into();
+                    let mut config_snapshot = ConfigSnapshot::default();
+                    let dependent_flow_key = if let FlowResult::DatasetCompact(_) = flow_result
+                        && flow.config_snapshot.compacting_rule.is_some()
+                        && flow
+                            .config_snapshot
+                            .compacting_rule
+                            .unwrap()
+                            .is_keep_metadata_only()
+                    {
+                        config_snapshot = flow.config_snapshot.clone();
+                        FlowKeyDataset::new(
+                            dependent_dataset_id.clone(),
+                            DatasetFlowType::HardCompacting,
+                        )
+                        .into()
+                    } else {
```

Comment:
Batching rule was a good condition for updates, if we enabled automatic transforms.
I'm not seeing the same requirement for compacting case - the derived have to be reset regardless of whether they have an automatic update configured. 

This needs some review of responsibilities - what is checked by Flows System, and what is programmed for each different flow type.

Reply:
Changed